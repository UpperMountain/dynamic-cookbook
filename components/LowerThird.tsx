import React from "react";
import { View } from "react-native";
import { LinearGradient } from "expo";
import { padding } from "../lib/theme";

export interface Props {
  children: React.ReactNode;
  third?: React.ReactNode;
}

export default function LowerThird({ children, third = null }: Props) {
  return (
    <View style={{ flex: 1 }}>
      {children}
      {third !== null && (
        <LinearGradient
          colors={[0, 0.5, 1].map((e: number) => `rgba(255,255,255,${e})`)}
          locations={[0, 0.2, 0.8]}
          style={{
            position: "absolute",
            bottom: 0,
            width: "100%",
            paddingTop: 1 * padding
          }}
        >
          {third}
        </LinearGradient>
      )}
    </View>
  );
}
