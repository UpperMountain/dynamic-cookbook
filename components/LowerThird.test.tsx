import React from "react";
import { Text } from "react-native";
import LowerThird from "./LowerThird";
import renderer from "react-test-renderer";

it("should render without exploding", () => {
  renderer
    .create(
      <LowerThird third={<Text>third</Text>}>
        <Text>contents</Text>
      </LowerThird>
    )
    .unmount();
});

it("should pass snapshot tests", () => {
  const cmp = renderer.create(
    <>
      <LowerThird third={<Text>third</Text>}>
        <Text>contents</Text>
      </LowerThird>
      <LowerThird>
        <Text>contents</Text>
      </LowerThird>
    </>
  );

  expect(cmp.toJSON()).toMatchSnapshot();
  cmp.unmount();
});
