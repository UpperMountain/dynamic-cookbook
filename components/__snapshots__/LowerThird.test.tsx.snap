// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass snapshot tests 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      contents
    </Text>
    <ExponentLinearGradient
      colors={
        Array [
          16777215,
          2164260863,
          4294967295,
        ]
      }
      locations={
        Array [
          0,
          0.2,
          0.8,
        ]
      }
      style={
        Object {
          "bottom": 0,
          "paddingTop": 20,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Text>
        third
      </Text>
    </ExponentLinearGradient>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      contents
    </Text>
  </View>,
]
`;
